canary/generator/transformators/transformers_categorical.py

```diff
@@ -0,0 +1,94 @@
+from abc import ABC
+from copy import deepcopy
+
+import numpy as np
+
+from .base import Transformer
+
+
+class TransformerCategorical(Transformer, ABC):
+    def __init__(self):
+        super(TransformerCategorical, self).__init__(['categorical', 'enumerated',
+                                                      'boolean', 'flag'])
+
+
+class AddToSomeBucketTransformer(TransformerCategorical):
+    def __init__(self, percentage=None, bucket_number=None, change_date=None):
+        """anomaly, works on buckets!!!"""
+        super(AddToSomeBucketTransformer, self).__init__()
+        self.is_bucket = False if bucket_number is None else True
+        self.is_percentage = False if percentage is None else True
+        self.is_date = False if change_date is None else True
+        self.percentage = percentage
+        self.bucket_number = bucket_number
+        self.change_date = change_date
+
+    def transform(self, bucket_dict, y):
+        bucket_dict_copy = deepcopy(bucket_dict)
+        y_copy = deepcopy(y)
+        if not self.is_date:
+            self.change_date = np.random.choice(list(bucket_dict_copy['data'].keys()))
+        hist = bucket_dict_copy['data'][self.change_date]
+        if not self.is_bucket:
+            self.bucket_number = np.random.choice(
+                list(range(len(hist))))
+        if not self.percentage:
+            self.percentage = np.random.uniform(0.2, 0.6)
+
+        hist[self.bucket_number] += self.percentage
```

Comment:
Are we adding the percentage value to the value of the bucket, here? Shouldn't it be something like `hist[self.bucket_number] += hist[self.bucket_number] * self.percentage`?

Reply:
If we do like that, there is a great probability we're not adding anything at all ;) I know it's not a nice solution, but I don't know any better... Maybe the name is not really suitable. What about `self.val_to_add`?